packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -669,6 +702,48 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
         emit RewardClaimFinalized(msg.sender, rewardsToClaim);
     }
 
+    /**
+     * @inheritdoc IAssetManager
+     */
+    function claimKghReward(address validator, uint256[] calldata tokenIds) external {
+        if (validator == address(0)) revert ZeroAddress();
+        if (tokenIds.length == 0) revert NotAllowedZeroInput();
+
+        Vault storage vault = _vaults[validator];
+        KghDelegator storage kghDelegator = vault.kghDelegators[msg.sender];
+        if (kghDelegator.kghNum != uint128(tokenIds.length)) revert InvalidTokenIdsInput();
+
+        uint128 claimedRewards = _claimBoostedReward(validator, msg.sender);
+
+        // Burn kroShares to claim the base reward.
+        uint128 kroSharesToBurn;
+        for (uint256 i = 0; i < tokenIds.length; ) {
+            if (kghDelegator.delegatedAt[tokenIds[i]] == 0) revert InvalidTokenIdsInput();
+            uint128 kroShares = kghDelegator.kroShares[tokenIds[i]] -
+                _convertToKroShares(validator, KGH_MANAGER.totalKroInKgh(tokenIds[i]));
+
+            unchecked {
+                kroSharesToBurn += kroShares;
+
+                ++i;
+            }
+        }
+
+        uint128 kghBaseReward = _convertToKroAssets(validator, kroSharesToBurn);
+
+        unchecked {
+            claimedRewards += kghBaseReward;
+            vault.asset.totalKroShares -= kroSharesToBurn;
+            vault.asset.totalKro -= kghBaseReward;
+        }
+
+        if (claimedRewards == 0) revert InsufficientAsset();
```

Comment:
How about move this line in L717? Because if totalBoostedReward is 0, we don't need to process anymore.

Reply:
Because if `totalBoostedReward` is zero, it means that there is no reward to claim for base reward as well, right?

Reply:
Yep that's right